annorxiver_modules/corpora_comparison_helper.py

```diff
@@ -0,0 +1,411 @@
+                for row in reader
+            })
+
+    return global_word_counter
+
+def dump_to_dataframe(count_dict, file_name):
+    """
+    This function outputs the counter data object in
+    the form of a tsv file. This was created to avoid
+    the memory issue PMC creates.
+    
+    Arguments:
+        count_dict - the counter object to be outputted
+        file_name - the name of the file for the output to be written to
+    """
+    # Avoids memory issues
+    with open(file_name, "w") as tsvfile:
+        reader = csv.DictWriter(
+            tsvfile, 
+            fieldnames=["lemma", "pos_tag", "dep_tag", "count"], 
+            delimiter="\t"
+        )
+
+        reader.writeheader()
+        for item in tqdm_notebook(count_dict):
+            reader.writerow({
+                "lemma":item[0],
+                "pos_tag":item[1],
+                "dep_tag":item[2],
+                "count":count_dict[item]
+            })
+
+def get_term_statistics(corpus_one, corpus_two, freq_num, psudeocount=1):
+    """
+    This function is designed to perform the folllowing calculations:
+        - log likelihood of contingency table
+        - log odds ratio
+        
+    keywords:
+        corpus_one - a dataframe object with terms and counts 
+        corpus_two - a datafram object with terms and counts
+        freq_num - number of most common words to use from both corpora
+    """
+    term_list = (
+        set(
+            corpus_one
+            .sort_values("count", ascending=False)
+            .head(freq_num)
+            .lemma
+            .values
+        )
+        |
+        set(
+            corpus_two
+            .sort_values("count", ascending=False)
+            .head(freq_num)
+            .lemma
+            .values
+        )
+    )
+    
+    corpus_one_total = corpus_one['count'].sum()
+    corpus_two_total = corpus_two['count'].sum()
+    
+    term_data = []
+    for term in tqdm_notebook(term_list):
+
+        corpus_one_term_count = (
+            corpus_one.query(f"lemma=='{term}'")['count'].values[0]
+            if term in corpus_one.lemma.tolist()
+            else 0
+        )
+        
+        corpus_two_term_count = (
+            corpus_two.query(f"lemma=='{term}'")['count'].values[0]
+            if term in corpus_two.lemma.tolist()
+            else 0
+        )
+        
+        observed_contingency_table = np.array([
+            [corpus_one_term_count, corpus_two_term_count],
+            [corpus_one_total, corpus_two_total]
+        ])
+
+        # Log Likelihood
+
+        ## add psudeocount to prevent log(0)
+        observed_contingency_table += psudeocount
+
+        a, b, c, d = (
+            observed_contingency_table[0][0],
+            observed_contingency_table[0][1],
+            observed_contingency_table[1][0],
+            observed_contingency_table[1][1]
+        )
+
+        # Obtained from (Kilgarriff, 2001) - Comparing Corpora
+        LL = lambda a,b,c,d: 2*(
```

Comment:
I'm not sure if I follow what this likelihood is telling you?

Reply:
> I'm not sure if I follow what this likelihood is telling you? 

The likelihood tells you how well does your model fit the data. By itself it is meaningless. You need to have at least two likelihood values to have any meaning. The bigger the value the better the fit.